library/core/src/result.rs

```diff
@@ -1668,7 +1668,7 @@ impl<T, E> ops::TryV2 for Result<T, E> {
 
 #[unstable(feature = "try_trait_v2", issue = "84277")]
 impl<T, E, F: From<E>> ops::FromResidual<Result<convert::Infallible, E>> for Result<T, F> {
-    #[inline]
+    #[inline(always)]
     fn from_residual(residual: Result<convert::Infallible, E>) -> Self {
         match residual {
             Err(e) => Err(From::from(e)),
```

Comment:
> The `from` here is the only one of these about which I'm potentially skeptical. If, say, `anyhow::Error: From<&str>` gets inlined into `from_residual` then we might not want to force that to then be inlined into every caller -- especially since this is the error path, where a function call is more tolerable.

Yeah, I also wasn't sure about this one, but I forgot that there was the possibility of something being inlined into `from_residual` that would then be inlined into the caller. I guess let's wait for the perf results and then consider removing this particular change :)